edb/server/server.py

```diff
@@ -906,33 +909,63 @@ async def _restart_servers_new_addr(self, nethosts, netport):
             for host, srv in self._servers.items():
                 if host == ADMIN_PLACEHOLDER or host in nethosts:
                     servers[host] = srv
+                elif host in ['::', '0.0.0.0']:
+                    servers_to_stop_early.append(srv)
                 else:
+                    if has_ipv4_wc:
+                        try:
+                            ipaddress.IPv4Address(host)
+                        except ValueError:
+                            pass
+                        else:
+                            servers_to_stop_early.append(srv)
+                            continue
+                    if has_ipv6_wc:
+                        try:
+                            ipaddress.IPv6Address(host)
+                        except ValueError:
+                            pass
+                        else:
+                            servers_to_stop_early.append(srv)
+                            continue
                     servers_to_stop.append(srv)
             admin = False
         else:
             hosts_to_start = nethosts
             servers_to_stop = self._servers.values()
             admin = True
 
+        if servers_to_stop_early:
+            await self._stop_servers_with_logging(servers_to_stop_early)
+
         if hosts_to_start:
-            new_servers, *_ = await self._start_servers(
-                hosts_to_start,
-                netport,
-                admin=admin,
-            )
-            servers.update(new_servers)
+            try:
+                new_servers, *_ = await self._start_servers(
```

Comment:
Ideally we shouldn't fail all servers because one of them failed to start.  This can be fixed in a follow up PR, though.

Reply:
I think you already did that in #3210?

Reply:
Ah, right :-)